src/compatibility.h

```diff
@@ -0,0 +1,47 @@
+#ifndef CATA_COMPATIBILITY_H
+#define CATA_COMPATIBILITY_H
+
+//--------------------------------------------------------------------------------------------------
+// HACK:
+// std::to_string is broken on MinGW (as of 13/01/2015), but is fixed in MinGW-w64 gcc 4.8.
+// See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=52015
+// This is a minimal workaround pending the issue begin solved (if ever).
+// For proper linking these must be declared inline here, or moved to a cpp file.
+//--------------------------------------------------------------------------------------------------
+#include <string>
+
+#define CATA_GCC_VER (__GNUC__ * 10000) + (__GNUC_MINOR__ * 100) + (__GNUC_PATCHLEVEL__)
+
+#if defined(__MINGW32__) && !defined(__MINGW64__)
+#   define CATA_NO_CPP11_STRING_CONVERSIONS
+#elif defined(__MINGW64__) && (CATA_GCC_VER < 40800)
+#   define CATA_NO_CPP11_STRING_CONVERSIONS
+#endif
+
+#if defined(CATA_NO_CPP11_STRING_CONVERSIONS)
+#include <cstdio>
+#include <limits>
+
+namespace std {
```

Comment:
You aren't; hence workaround / hack. It's undefined behavior in that it could conflict with god knows what, but this is targeted at a very specific platform only.

There's nothing magical or forbidden about putting something in std::, the standard just says there's no guarantee it's going to do what you think everywhere (every platform), or that it could break in some future revision of the standard.

Here I think we're pretty safe; all the more so considering mingw32 isn't a high profile target platform AFAIK.